### PR TITLE
test(many): replace ionic buttons in e2e tests with native html buttons

### PR DESCRIPTION
**Pull request by brandyscarney**
Issue number: internal

---------

## What is the current behavior?
The `ion-button` component is used in several tests to navigate or show overlays. This causes screenshot diffs in unrelated tests any time the UI of the `ion-button` is updated. 

## What is the new behavior?
Removes the `ion-button` elements from unrelated tests. 

Did not remove the `ion-button`s from the following tests:
- All `ion-button`s in an `ion-buttons` component
- An `ion-button` inside of a menu
- breadcrumbs/test/basic (uses a clear button in a list header, needs to be moved)
- input/test/slot
- item/test/buttons
- item/test/colors
- item/test/dividers
- item/test/inputs
- item/test/media
- list-header/test/basic
- ripple-effect/test/basic
- router/test/basic
- router/test/guards
- router-outlet/test/basic
- select/test/slot
- textarea/test/slot

Updates the icon/basic test to use the right icon names by comparing against the v3 names: https://ionicframework.com/docs/v3/ionicons/

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/icon/test/basic/index.html [hunk 1]
```diff
@@ -143,7 +152,7 @@ <h3>mode: md</h3>
       dynamicHomeIcon.name = 'home';
       dynamicHomeIcon.isActive = false;
 
-      icons = ['home', 'star', 'ios-alert', 'ios-alert-outline', 'md-alert', 'logo-apple'];
+      icons = ['home', 'star', 'alert-circle', 'alert-circle-outline', 'alert-circle-sharp', 'logo-apple'];
```
Comment (brandyscarney) [inline core/src/components/icon/test/basic/index.html:155]: I updated this to use the latest Ionicon names by comparing the older names with the v3 Ionicons to see what it should be now: https://ionicframework.com/docs/v3/ionicons

File: core/scripts/testing/styles.css [hunk 2]
```diff
@@ -48,3 +48,15 @@ html.ios.ios {
   --ion-font-family: -apple-system, BlinkMacSystemFont, "iosTestingFont", sans-serif;
   font-family: -apple-system, BlinkMacSystemFont, "iosTestingFont", sans-serif;
 }
+
+button {
+  display: block;
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (brandyscarney) [reply core/scripts/testing/styles.css]: Spoke with the team during office hours. We decided to update the button to be `inline-block` by default and then add a class called `expand` which replaces the old `expand="block"` behavior that using an `ion-button` had.

File: core/src/components/content/test/basic/index.html [hunk 3]
```diff
@@ -30,16 +30,16 @@
           <ion-content id="content-part">
             <f style="height: 50%"></f>
             <p>
-              <ion-button onclick="toggleDisplay('header')" color="secondary">Toggle header</ion-button>
-              <ion-button onclick="toggleDisplay('footer')" color="secondary">Toggle footer</ion-button>
-              <ion-button onclick="toggleDisplay('toolbar2')" color="secondary">Toggle 2nd toolbar</ion-button>
+              <button onclick="toggleDisplay('header')" color="secondary">Toggle header</button>
```
Comment (sean-perkins) [inline core/src/components/content/test/basic/index.html:33]: Can the `color` properties be removed now that these are `button` elements?

File: core/src/components/searchbar/test/basic/index.html [hunk 4]
```diff
@@ -98,27 +98,15 @@ <h5 class="ion-padding-start ion-padding-top">Search - Animated and No Cancel</h
...
-        </div>
-        <div class="ion-padding">
-          <ion-button expand="block" color="secondary" onClick="toggleAttr()">Toggle Attribute</ion-button>
-        </div>
-
-        <div class="ion-padding-horizontal">
-          <ion-button expand="block" color="secondary" onClick="toggleDisabled()">Toggle disabled</ion-button>
-        </div>
+        <button class="expand" onClick="toggleProp()">Toggle Property</button>
+        <button class="expand" color="secondary" onClick="toggleAttr()">Toggle Attribute</button>
```
Comment (sean-perkins) [inline core/src/components/searchbar/test/basic/index.html:108]: Similar question here around the `color` attribute.

File: core/src/components/segment/test/basic/index.html [hunk 5]
```diff
@@ -215,15 +215,9 @@
...
-          <ion-button expand="block" onClick="toggleDisabled()">Toggle Disabled</ion-button>
-        </div>
-        <div class="ion-padding">
-          <ion-button expand="block" color="secondary" onClick="toggleValue()">Toggle Value</ion-button>
-        </div>
-        <div class="ion-padding-horizontal">
-          <ion-button expand="block" color="tertiary" onClick="toggleSwipeGesture()">Toggle Swipe Gesture</ion-button>
-        </div>
+        <button class="expand" onClick="toggleDisabled()">Toggle Disabled</button>
+        <button class="expand" color="secondary" onClick="toggleValue()">Toggle Value</button>
```
Comment (sean-perkins) [inline core/src/components/segment/test/basic/index.html:219]: & here for the `color` attribute.

File: core/src/components/tabs/test/basic/index.html [hunk 6]
```diff
@@ -30,8 +42,8 @@
           </ion-header>
           <ion-content class="ion-padding">
             <h1>Tab One</h1>
-            <ion-button expand="block" onclick="updateBadgeCount()">Update Badge Count</ion-button>
-            <ion-button color="secondary" expand="block" onclick="updateBadgeColor()">Update Badge Color</ion-button>
+            <button class="expand" onclick="updateBadgeCount()">Update Badge Count</button>
+            <button class="expand" color="secondary" onclick="updateBadgeColor()">Update Badge Color</button>
```
Comment (sean-perkins) [inline core/src/components/tabs/test/basic/index.html:46]: & here for the `color` attribute.

**Review by sean-perkins — COMMENTED** [review]
One question that applies to a few places around the `color` attribute on the `button` element.

File: core/src/components/content/test/basic/index.html (hunk 3 above)
Comment (brandyscarney) [reply core/src/components/content/test/basic/index.html]: Removed in https://github.com/ionic-team/ionic-framework/pull/29422/commits/74d7dd7719335983a82364ba10fb54f9136522ea

File: core/src/components/searchbar/test/basic/index.html (hunk 4 above)
Comment (brandyscarney) [reply core/src/components/searchbar/test/basic/index.html]: Removed in https://github.com/ionic-team/ionic-framework/pull/29422/commits/74d7dd7719335983a82364ba10fb54f9136522ea

File: core/src/components/segment/test/basic/index.html (hunk 5 above)
Comment (brandyscarney) [reply core/src/components/segment/test/basic/index.html]: Removed in https://github.com/ionic-team/ionic-framework/pull/29422/commits/74d7dd7719335983a82364ba10fb54f9136522ea

File: core/src/components/tabs/test/basic/index.html (hunk 6 above)
Comment (brandyscarney) [reply core/src/components/tabs/test/basic/index.html]: Removed in https://github.com/ionic-team/ionic-framework/pull/29422/commits/74d7dd7719335983a82364ba10fb54f9136522ea

**Review by sean-perkins — APPROVED** [review]
Great work!

**Pull request merged**